crates/walrus-core/src/encoding/quilt_encoding.rs

```diff
@@ -516,13 +1122,105 @@ pub struct QuiltEncoderV1<'a> {
 
 impl<'a> QuiltEncoderV1<'a> {
     /// Creates a new [`QuiltEncoderV1`] from a encoding config and a set of blobs.
-    pub fn new(config: EncodingConfigEnum<'a>, blobs: &'a [BlobWithIdentifier<'a>]) -> Self {
+    pub fn new(config: EncodingConfigEnum<'a>, blobs: &'a [QuiltStoreBlob<'a>]) -> Self {
         Self {
             blobs,
             config,
             span: tracing::span!(Level::ERROR, "QuiltEncoderV1"),
         }
     }
+
+    /// Returns the extension bytes of the blob.
+    pub fn get_header_and_extension_bytes(blob: &QuiltStoreBlob) -> Result<Vec<u8>, QuiltError> {
+        let mut identifier_bytes = Vec::new();
+        let mut extension_bytes = Vec::new();
+        let mut header = BlobHeaderV1::default();
+
+        let identifier_size =
+            u16::try_from(bcs::serialized_size(&blob.identifier).map_err(|e| {
+                QuiltError::Other(format!("Failed to serialize identifier: {}", e))
+            })?)
+            .map_err(|e| {
+                QuiltError::Other(format!("Failed to convert identifier size to u16: {}", e))
+            })?;
```

Comment:
Nit: We have almost the same code above, can we maybe reuse the `compute_encoded_blob_size` here? Also, can we have a separate error variant for invalid identifiers?